### PR TITLE
Fix FN_PUBLIC_LB_URL when TLS is enabled

### PR DESCRIPTION
**Pull request by devcamcar**

### CODE REVIEW

**Review by denismakogon — CHANGES_REQUESTED** [review]
Thanks, you can make it slightly simpler without duplicating the whole URL definition just for the one character.
Take a look at FN_DB_URL (create environment var that will contain the whole line without protocol prefix, then use it as a reference  `https://($FN_PUBLIC_LB_ENDPOINT)` )

File: fn/templates/fn-deployment.yaml [hunk 1]
```diff
@@ -57,7 +57,11 @@ spec:
           - name: FN_NODE_TYPE
             value: "api"
           - name: FN_PUBLIC_LB_URL
+            {{- if .Values.tls.enabled }}
+            value: https://{{ .Release.Name }}.{{ .Values.fn_lb_runner.service.ingress_hostname }}:{{ .Values.fn_lb_runner.service.port }}
```
Comment (denismakogon) [inline fn/templates/fn-deployment.yaml]:
```yaml
- name: FN_PUBLIC_LB_SCHEMA
{{- if .Values.tls.enabled }}
  value: https
{{ else }}
  value: http
{{- end }}
- name: FN_PUBLIC_LB_URL:
  value: $(FN_PUBLIC_LB_SCHEMA)://{{ .Release.Name }}.{{ .Values.fn_lb_runner.service.ingress_hostname }}:{{ .Values.fn_lb_runner.service.port }}
```

Comment (devcamcar) [reply fn/templates/fn-deployment.yaml]: Done by using templates now

File: fn/templates/fn-deployment.yaml [hunk 2]
```diff
@@ -58,9 +58,9 @@ spec:
             value: "api"
           - name: FN_PUBLIC_LB_URL
             {{- if .Values.tls.enabled }}
-            value: https://{{ .Release.Name }}.{{ .Values.fn_lb_runner.service.ingress_hostname }}:{{ .Values.fn_lb_runner.service.port }}
+            value: https://{{ template "fn.public_lb_url" . }}
```
Comment (denismakogon) [inline fn/templates/fn-deployment.yaml]: that's too complicated and hard to search for references,
here's what you need to do:
```yaml
- name: FN_PUBLIC_LB_SCHEMA
{{- if .Values.tls.enabled }}
  value: https
{{ else }}
  value: http
{{- end }}
- name: FN_PUBLIC_LB_URL:
  value: $(FN_PUBLIC_LB_SCHEMA)://{{ .Release.Name }}.{{ .Values.fn_lb_runner.service.ingress_hostname }}:{{ .Values.fn_lb_runner.service.port }}
```
that's it, looks similar to what we have for FN_DB_URL (clear and simple)

FYI, templates are for more complex logic, but not for the one-line templates.

**Review by denismakogon — CHANGES_REQUESTED** [review]

File: fn/templates/fn-deployment.yaml (hunk 2 above)
Comment (denismakogon) [reply fn/templates/fn-deployment.yaml]: you know, actually, ignore my comment, these two fixes doing the same but in a different manner, which is totally fine.

**Review by denismakogon — APPROVED** [review]

**Pull request merged**